src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import java.util.Map;
+
+import static org.elasticsearch.script.ScriptService.ScriptType;
+
+public class Script {
+
+    public final ScriptType type;
+    public final String lang;
+    public final String key;
+    public final ScriptContext context;
+    public final Map<String, Object> params;
+    public final Object compiled;
+
+    public Script(String lang, String key, ScriptType type, ScriptContext context) {
+        this(lang, key, type, context, null, null);
+    }
+
+    public Script(String lang, String key, ScriptType type, ScriptContext context, Map<String, Object> params) {
+        this (lang, key, type, context, params, null);
+    }
+
+    public Script(String lang, String key, ScriptType type, ScriptContext context, Map<String, Object> params, Object compiled) {
+        this.type = type;
+        this.lang = lang;
+        this.key = key;
+        this.context = context;
+        this.params = params;
+        this.compiled = compiled;
+    }
+
+    public Script(Script script, ScriptContext context) {
+        this(script.lang, script.key, script.type, context, script.params, script.compiled);
+    }
+
+    public Script(Script script, Map<String, Object> params) {
+        this(script.lang, script.key, script.type, script.context, params, script.compiled);
+    }
+
+    public Script(Script script, String lang, Object compiled) {
+        this(lang, script.key, script.type, script.context, script.params, compiled);
+    }
```

Comment:
Done.  Reduced to a single constructor using null values to pass in where appropriate.